test/extended/builds/docker_pullsecret.go

```diff
@@ -35,29 +35,16 @@ var _ = g.Describe("[builds][pullsecret][Conformance] docker build using a pull
 			err := oc.Run("create").Args("-f", buildFixture).Execute()
 			o.Expect(err).NotTo(o.HaveOccurred())
 
-			g.By("starting a test build")
-			out, err := oc.Run("start-build").Args("docker-build").Output()
-			fmt.Fprintf(g.GinkgoWriter, "\nstart-build output:\n%s\n", out)
-			o.Expect(err).NotTo(o.HaveOccurred())
-
-			g.By("expecting the build succeeds")
-			err = exutil.WaitForABuild(oc.REST().Builds(oc.Namespace()), "docker-build-1", exutil.CheckBuildSuccessFn, exutil.CheckBuildFailedFn)
-			if err != nil {
-				exutil.DumpBuildLogs("docker-build", oc)
-			}
+			g.By("starting a build")
+			br, err := exutil.RunBuildConfigAndWait(oc, "docker-build")
```

Comment:
I don't love "RunStartBuild" as a name, if we're going to clean this up, i'd suggest keeping the "StartBuild" function name and repurposing it.

And also change RunStartBuildConfigAndWait to "StartBuildConfigAndWait" i think.

(and should StartBuild become StartBuildAndWait ?  are there cases where we don't want to wait?  How do both of these methods handle cases where I need to expect failure on the start-build operation?)

Also RunStartBuildConfigAndWait can actually take a build argument, no?  so it's not well named.

As I think about this more generally, I'm worried about the encapsulation of all this logic.  Test cases are still going to need to:
1) start a build and not wait (possibly, needed for the serialized build tests I believe)
2) start a build and expect failure (so asserting success is not always valid).

So keeping this as a separate set of methods (start a build/buildconfig, wait for a build, check success of a build) seems more logical to me.